Add a setting to control losslessly compressed IOSurface usage

smfr opened the pull request:
#### 6ea829cc8cc6d8d2261f70808ac17eb5700c12d1
<pre>
Add a setting to control losslessly compressed IOSurface usage
<a href="https://bugs.webkit.org/show_bug.cgi?id=296582">https://bugs.webkit.org/show_bug.cgi?id=296582</a>
<a href="https://rdar.apple.com/156935443">rdar://156935443</a>

Reviewed by Tim Horton.

Add the IOSurfaceLosslessCompressionEnabled setting. On Cocoa platforms,
this is only enabled when `HAVE(COREVIDEO_COMPRESSED_PIXEL_FORMAT_TYPES)`
and `HAVE(LOSSLESS_COMPRESSED_IOSURFACE_CG_SUPPORT)` are true
(i.e. iOS device, macOS), and when CASupportsFeature() says that
we can use it. `CA_FEATURE_CG_COMPRESSED_IOSURFACES` will be read from
an SPI header in future.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/PAL/pal/spi/cocoa/QuartzCoreSPI.h:
* Source/WebKit/Shared/Cocoa/WebPreferencesDefaultValuesCocoa.mm:
(WebKit::defaultIOSurfaceLosslessCompressionEnabled):
* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultIOSurfaceLosslessCompressionEnabled):
* Source/WebKit/Shared/WebPreferencesDefaultValues.h:

Canonical link: <a href="https://commits.webkit.org/297967@main">https://commits.webkit.org/297967@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5b20d2bf3b113842f676bb801b4a9695caaf6ec3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/113679 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/131/builds/33383 "Build was cancelled. Recent messages:OS: Sequoia (15.5), Xcode: 16.4") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/138/builds/23840 "Build was cancelled. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (cancelled)") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/119842 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/64455 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/115568 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/33991 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/41969 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86432 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/64455 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/116627 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/27123 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/102149 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/66785 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/26361 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20274 "Passed tests") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/63572 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/106137 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/96504 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20377 "Passed tests") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/123081 "Build was cancelled. Recent messages:Printed configuration; Running apply-patch; Checked out pull request; Compiled WebKit (cancelled)") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/112241 "Built successfully and passed tests") | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/40679 "Build was cancelled. Recent messages:OS: Sequoia (15.5), Xcode: 16.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (cancelled)") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/30363 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/95286 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/41070 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/98364 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/95042 "Passed tests") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24249 "Failed to push commit to Webkit repository") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40167 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17987 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/36894 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/40561 "Build was cancelled. Recent messages:OS: Sequoia (15.5), Xcode: 16.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (cancelled)") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/46070 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/136450 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/40207 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/36570 "Found 2 new JSC stress test failures: stress/builtin-function-is-construct-type-none.js.dfg-eager, stress/dont-link-virtual-calls-on-compiler-thread.js.no-llint (failure)") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/43508 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/41976 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebKit/Shared/Cocoa/WebPreferencesDefaultValuesCocoa.mm:
```diff
@@ -142,6 +143,17 @@ bool defaultHostedBlurMaterialInMediaControlsEnabled()
+bool defaultIOSurfaceLosslessCompressionEnabled()
+{
+#if HAVE(COREVIDEO_COMPRESSED_PIXEL_FORMAT_TYPES)
+#define CA_FEATURE_CG_COMPRESSED_IOSURFACES 15
```
do you want a WK here? or are they not using the same name?

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/02774b6d23ab85cfd9771bfe0e643c12ef7275c1. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48620

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/02774b6d23ab85cfd9771bfe0e643c12ef7275c1. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48620

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5b20d2bf3b113842f676bb801b4a9695caaf6ec3. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48620

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5b20d2bf3b113842f676bb801b4a9695caaf6ec3. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48620

smfr committed:
Add a setting to control losslessly compressed IOSurface usage
https://bugs.webkit.org/show_bug.cgi?id=296582
rdar://156935443

Reviewed by Tim Horton.

Add the IOSurfaceLosslessCompressionEnabled setting. On Cocoa platforms,
this is only enabled when `HAVE(COREVIDEO_COMPRESSED_PIXEL_FORMAT_TYPES)`
and `HAVE(LOSSLESS_COMPRESSED_IOSURFACE_CG_SUPPORT)` are true
(i.e. iOS device, macOS), and when CASupportsFeature() says that
we can use it. `CA_FEATURE_CG_COMPRESSED_IOSURFACES` will be read from
an SPI header in future.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/PAL/pal/spi/cocoa/QuartzCoreSPI.h:
* Source/WebKit/Shared/Cocoa/WebPreferencesDefaultValuesCocoa.mm:
(WebKit::defaultIOSurfaceLosslessCompressionEnabled):
* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultIOSurfaceLosslessCompressionEnabled):
* Source/WebKit/Shared/WebPreferencesDefaultValues.h:

Canonical link: https://commits.webkit.org/297967@main

smfr commented:
This landed.
